netdeploy/remote/deployedNetwork.go

```diff
@@ -283,9 +338,424 @@ func (cfg DeployedNetwork) BuildNetworkFromTemplate(buildCfg BuildConfig, rootDi
+	if rewardsPool.Address == "" {
+		return fmt.Errorf("error finding source rewards ppol address")
+
+	}
+
+	feeSink := getGenesisAlloc("FeeSink", genesis.Allocation)
+	if feeSink.Address == "" {
+		return fmt.Errorf("error finding fee sink address")
+
+	}
+	src, err := basics.UnmarshalChecksumAddress(srcWallet.Address)
+
+	poolAddr, err := basics.UnmarshalChecksumAddress(rewardsPool.Address)
+	if err != nil {
+		return err
+	}
+	sinkAddr, err := basics.UnmarshalChecksumAddress(feeSink.Address)
+	if err != nil {
+		return err
+	}
+
+	//initial state
+	bootstrappedNet.nAssets = fileCfgs.GeneratedAssetsCount
+	bootstrappedNet.nApplications = fileCfgs.GeneratedApplicationCount
+	//bootstrappedNet.nApplications = 0
+	bootstrappedNet.txState = protocol.PaymentTx
+	bootstrappedNet.roundTrxCnt = fileCfgs.RoundTransactionsCount
+	bootstrappedNet.round = basics.Round(0)
+
+	params := config.Consensus[protocol.ConsensusCurrentVersion]
+	minAccounts := accountsNeeded(fileCfgs.GeneratedApplicationCount, fileCfgs.GeneratedAssetsCount, params)
+	nAccounts := fileCfgs.GeneratedAccountsCount
+	if minAccounts > nAccounts {
+		bootstrappedNet.nAccounts = minAccounts
+	} else {
+		bootstrappedNet.nAccounts = nAccounts
+	}
+
+	accounts[poolAddr] = basics.MakeAccountData(basics.NotParticipating, rewardsPool.State.MicroAlgos)
+	accounts[sinkAddr] = basics.MakeAccountData(basics.NotParticipating, feeSink.State.MicroAlgos)
+	//fund src account with enough funding
+	bootstrappedNet.fundPerAccount = basics.MicroAlgos{Raw: uint64(bootstrappedNet.nAssets) * params.MinBalance * 2}
+	totalFunds := srcWallet.State.MicroAlgos.Raw + bootstrappedNet.fundPerAccount.Raw*bootstrappedNet.nAccounts + bootstrappedNet.fundPerAccount.Raw*bootstrappedNet.roundTrxCnt*fileCfgs.NumRounds
+	accounts[src] = basics.MakeAccountData(basics.Online, basics.MicroAlgos{Raw: totalFunds})
+
+	bootstrappedNet.poolAddr = poolAddr
+	bootstrappedNet.sinkAddr = sinkAddr
+
+	//gensis block
+	initState, err := generateInitState(accounts, &bootstrappedNet)
+	if err != nil {
+		return err
+	}
+	localCfg := config.GetDefaultLocal()
+	localCfg.Archival = true
+	localCfg.CatchpointTracking = -1
+	localCfg.LedgerSynchronousMode = 0
+	log := logging.NewLogger()
+	l, err := ledger.OpenLedger(log, genesisFolder+"/bootstrapped", false, initState, localCfg)
+	if err != nil {
+		return err
+	}
+	l.WaitForCommit(0)
+
+	//create accounts, apps and assets
+	prev, _ := l.Block(l.Latest())
+	err = generateAccounts(src, fileCfgs.RoundTransactionsCount, prev, l, &bootstrappedNet, params)
+	if err != nil {
+		return err
+	}
+
+	//create more transactions
+	prev, _ = l.Block(l.Latest())
+	for i := uint64(bootstrappedNet.round); i < fileCfgs.NumRounds; i++ {
+		bootstrappedNet.round++
+		blk, _ := createBlock(src, prev, fileCfgs.RoundTransactionsCount, &bootstrappedNet, params)
+		blk.BlockHeader.TimeStamp += int64(crypto.RandUint64() % 100 * 1000)
+		err = l.AddBlock(blk, agreement.Certificate{Round: bootstrappedNet.round})
```

Comment:
I wonder if there are options for adding "pre-validated" blocks in batches in order to economize on signature validations.

Reply:
I don't see an option to add blocks in batches, but AddBlock() has validate set to false, so it doesn't actually do signature validation during eval().